frontend: handle zero amount invoices when sending with lightning

thisconnect opened the pull request:
Added a new step edit-invoice to allow specifying a custom amount if no amount was given or amount is 0.

Other change:
- back button on confirm screen goes now back to the lighting view was going back to the QR scanner

dangeross commented:
> It seems as custom description / memo is not supported, added TODO
> (if that is even possible to set).
You normally set the description / memo when creating the invoice (receiver side), not when paying the invoice

thisconnect commented:
> You normally set the description / memo when creating the invoice (receiver side), not when paying the invoice

Thanks, I expected that it's not possible. I removed the TODO and updated the commit message, thanks @dangeross

thisconnect commented:
@Beerosagos PTAL

Beerosagos commented on frontends/web/src/routes/lightning/send/send.tsx:
```diff
@@ -220,9 +277,13 @@ export const Send = () => {
+        await postSendPayment({
+          bolt11: paymentDetails.invoice.bolt11,
+          // amountMsat is optional, if amount is missing the UI shows edit-invoice step for the user to enter a custom amountMsat which is passed here
+          amountMsat: customAmount
```
AFAIK amountMsat unit is milli satoshi. The amount should be multiplied * 1000

thisconnect replied:
oups 😇

Beerosagos commented:
> I sometimes got this error, but at least once it went through
> 
> ![Screen Shot 2024-02-22 at 12 56 12](https://private-user-images.githubusercontent.com/546900/306992863-0053957c-e816-4f21-9dbd-74ba33b7412a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDg5NTY4NDUsIm5iZiI6MTcwODk1NjU0NSwicGF0aCI6Ii81NDY5MDAvMzA2OTkyODYzLTAwNTM5NTdjLWU4MTYtNGYyMS05ZGJkLTc0YmEzM2I3NDEyYS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMjI2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDIyNlQxNDA5MDVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wY2M4YzlmYTI0ZDgwYjFjMWU0MmQxYTYyZTY1ZTIzOGNmZjkwOTI0NDE5ZmNiMjliZjQzZjg1ZGZjZDFjNjBjJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.jeAnBr1PIUHsGPnjdN8R_OXAgjmcuIffzFBnmeP8D9c)
> 
> in phoenix: "1 incoming payment recently rejected"
> 
> reason: "The fee was 2976 sat which is more than 50% of the amount"
> 
> I have 600k inbound liquidity but only 2k outbound

From my tests, the reason of this error is the missing conversion to milli sats: Setting 1000 in the input seems to successfully send a 1sat transaction

thisconnect commented on frontends/web/src/routes/lightning/send/send.tsx:
```diff
@@ -138,6 +141,53 @@ const SendWorkflow = ({
+                value={customAmount ? `${customAmount}` : ''}
+                autoFocus
+              />
+              <Input
```
I kept as readonly input so that the user can read the memo/description

thisconnect commented:
rebased

thisconnect commented:
@Beerosagos PTAL

Beerosagos reviewed (approved):
partially tested ACK
